baremetal/metal3data_manager.go

```diff
@@ -761,8 +762,8 @@ func (m *DataManager) getAddressFromPool(ctx context.Context, poolName string,
 	return addresses, false, nil
 }
 
-// releaseAddressFromPool removes the owner reference on existing referenced
-// Metal3IPPool objects.
+// releaseAddressFromPool deletes the IPClaim owned by this Metal3Data found in the IPPool.
+// Returns map of poolName => deleted, if it should requeue and error.
```

Comment:
I am not sure if this comment is clear, maybe we need to rephrase a bit?

Reply:
Updated, PTAL!